lightkurve/collection.py

```diff
@@ -0,0 +1,102 @@
+from abc import abstractmethod, ABCMeta
+
+__all__ = ['Collection']
+
+class Collection:
+    """Abstract Base Class for LightCurveCollections, TPFCollections
+
+    Attributes
+    ----------
+    data: array
+        List of data objects.
+    k_id: dictionary
+        Mapping target to index in self.data
+    """
+    __metaclass__ = ABCMeta #Needs to be set for Python 2.x to work properly
+    def __init__(self, data):
+        self.data = data
+        self.k_id = self.assign_hash_values()
+
+    def assign_hash_values(self):
```

Comment:
Nit-picking: let's make this a hidden method because users shouldn't have to use it.  Python syntax for this is underscore prefix, i.e. rename the function to `_assign_hash_values`  (or even better `_update_targetid_map`?)